core/src/main/java/io/questdb/cairo/sql/PageAddressCache.java

```diff
@@ -50,11 +57,21 @@ public void add(int frameIndex, @Transient PageFrame frame) {
             return; // The page frame is already cached
         }
         for (int columnIndex = 0; columnIndex < columnCount; columnIndex++) {
-            pageAddresses.add(frame.getPageAddress(columnIndex));
+            final long pageAddress = frame.getPageAddress(columnIndex);
+            pageAddresses.add(pageAddress);
             int varSizeColumnIndex = varSizeColumnIndexes.getQuick(columnIndex);
             if (varSizeColumnIndex > -1) {
-                auxPageAddresses.add(frame.getIndexPageAddress(columnIndex));
-                pageSizes.add(frame.getPageSize(columnIndex));
+                final long auxPageAddress = frame.getIndexPageAddress(columnIndex);
+                auxPageAddresses.add(auxPageAddress);
+                final long pageSize = frame.getPageSize(columnIndex);
+                pageSizes.add(pageSize);
+                final int columnFrameIndex = columnCount * frameIndex + columnIndex;
+                pageLimits.extendAndSet(columnFrameIndex, pageAddress + pageSize);
```

Comment:
this is too elaborate, you can just use `pageLimits.add()` and not calculate the index. These lists are all packaged there same. `pageSizes` and `pageLimits` have same structure, they need to be populated the same way